changelog/unreleased/issue-4585

```diff
@@ -0,0 +1,17 @@
+Enhancement: Limit repository size to a predefined maximum size during vackup
+
+Restic backup can now limit the repository size to a given maximum value.
+
+During backup, the current size is constantly monitored by watching its initial
+size plus the sum of "DataSizeInRepo" and "TreeSizeInRepo".
+
+The new packfiles are created by the current backup process, but have not been
+linked to a new snapshot. Therefore they are unreferenced, but
+are still occupying space in tge repository.
+
+I need help with shutting down the backup process properly, so the failed backup
```

Comment:
What's that doing in the changelog entry?

Reply:
Restic currently doesn't have the concept of a partial snapshot. Storing such a snapshot will currently cause the next backup run to re-read all files that didn't make it into the partial snapshot.

Adding a corresponding flag to the snapshot shouldn't be too hard. Similarly, the archiver and fileSaver could be adapted to stop reading new files. Then the whole backup process would quickly unwind itself and finish with a partial snapshot. (This should happen in a separate PR as partial snapshots are a useful feature in itself). Afterwards, we can think about repository size limits again. The main remaining complexity would be estimating how much space would still be necessary to write the remaining index and tree blobs when approaching the size limit.